valhalla/sif/dynamiccost.h

```diff
@@ -358,6 +363,58 @@ class DynamicCost {
                                           baldr::DateTime::get_tz_db().from_index(tz_index));
   }
 
+  inline static bool IsRestricted(const uint64_t restriction,
+                                  const uint64_t current_time,
+                                  const uint32_t tz_index,
+                                  baldr::AccessType access_type) {
+
+    if (access_type == baldr::AccessType::kTimedAllowed) {
```

Comment:
this also gets called a lot, i am wondering if the branches impact performance. i wouldnt suggest changing it for clarity unless performance is impacted by this PR. basically similar to the comment before i guess, can you do a quick gut check to see if performance is impacted by the refactoring.

Reply:
It's worth noting that this check is not new. It was just done in every single implementation of `Allowed`/`AllowedReverse` before as in https://github.com/valhalla/valhalla/blob/master/src/sif/autocost.cc#L395 . I merely moved the check to this single location.